finetune/base_models/tcn/featurizer.py

```diff
@@ -0,0 +1,138 @@
+    :param reuse: Should reuse be set within this scope.
+    :return: A dict containing;
+        embed_weights: the word embedding matrix.
+        features: The output of the featurizer_final state.
+        sequence_features: The output of the featurizer at each timestep.
+    """
+    initial_shape = tf.shape(X)
+    X = tf.reshape(X, shape=tf.concat(([-1], initial_shape[-2:]), 0))
+
+    with tf.variable_scope("model/featurizer", reuse=reuse):
+        embed_weights = tf.get_variable(
+            name="we",
+            shape=[encoder.vocab_size + config.max_length, config.n_embed_featurizer],
+            initializer=tf.random_normal_initializer(stddev=config.weight_stddev),
+        )
+
+        if config.train_embeddings:
+            embed_weights = dropout(embed_weights, config.embed_p_drop, train)
+        else:
+            embed_weights = tf.stop_gradient(embed_weights)
+
+        X = tf.reshape(X, [-1, config.max_length, 2])
+
+        # we remove positional embeddings from the model
+        h = embed(X[:, :, :1], embed_weights)
+
+        # keep track of the classify token
+        clf_token = encoder["_classify_"]
+        
+        class TemporalBlock(object):
+            def __init__(self, n_filters, kernel_size, dilation_rate, rate, scope, layer_num):
+                self.rate=rate
+                self.n_filters = n_filters
+                self.conv1 = tf.keras.layers.Conv1D(
+                    filters=n_filters,
+                    kernel_size=kernel_size,
+                    padding="same",
+                    activation=tf.nn.relu,
+                    dilation_rate=dilation_rate,
+                    kernel_initializer=tf.initializers.glorot_normal,
+                    name="conv1_" + str(layer_num),
+                )
+                self.conv2 = tf.keras.layers.Conv1D(
+                    filters=n_filters,
+                    kernel_size=kernel_size,
+                    padding="same",
+                    dilation_rate=dilation_rate,
+                    activation=tf.nn.relu,
+                    kernel_initializer=tf.initializers.glorot_normal,
+                    name="conv2_" + str(layer_num),
+                )
+                self.downsample = tf.keras.layers.Conv1D(filters=n_filters, kernel_size=1, padding="same")
+
+            def __call__(self, X):
+                conv1_out = self.conv1(X)
+                conv1_dropout = tf.nn.dropout(conv1_out, rate=self.rate)
+                conv2_out = self.conv2(conv1_dropout)
+                conv2_dropout = tf.nn.dropout(conv2_out, rate=self.rate)
+
+                # residuals
+                if X.get_shape()[-1] == self.n_filters:
+                    output = conv2_dropout + X
+                else:
+                    downsampled_out = self.downsample(X)
+                    output = conv2_dropout + downsampled_out
+
+                return output
+
+        with tf.variable_scope("crf"):
```

Comment:
Can we change this scope name? Its kinda confusing because we don't have the crf in here.